### PR TITLE
fix(nnx): make WeightNorm.scales learnable Param instances (fixes #5577)

### LINKED ISSUE #5577: The scales field of the NNX WeightNorm module isn't learnable.
It seems that `nnx.WeightNorm.scales` field isn't an `nnx.Param` and therefore won't be trainable, unlike what is specified it should be in the original weight normalisation paper: [Weight Normalization: A Simple Reparameterization to Accelerate Training of Deep Neural Networks](https://arxiv.org/abs/1602.07868).

### Problem you have encountered:
```python
In [7]: rngs = nnx.Rngs(42)

In [8]: wn = nnx.WeightNorm(nnx.Linear(3, 1, rngs=rngs), use_scale=True, rngs=rngs)

In [9]: wn.scales
Out[9]: {('kernel',): Array([1.], dtype=float32)}

In [10]: nnx.state(wn)
Out[10]:
State({
  'layer_instance': {
    'bias': Param( # 1 (4 B)
      value=Array([0.], dtype=float32)
    ),
    'kernel': Param( # 3 (12 B)
      value=Array([[ 0.04756278],
             [-0.30357933],
             [ 0.782016  ]], dtype=float32)
    )
  },
  'scales': {
    ('kernel',): Array([1.], dtype=float32)
  }
})

In [11]: nnx.state(wn, nnx.Param)
Out[11]:
State({
  'layer_instance': {
    'bias': Param( # 1 (4 B)
      value=Array([0.], dtype=float32)
    ),
    'kernel': Param( # 3 (12 B)
      value=Array([[ 0.04756278],
             [-0.30357933],
             [ 0.782016  ]], dtype=float32)
    )
  }
})
```

### What you expected to happen:
```python
In [15]: wn.scales[('kernel',)] = nnx.Param(wn.scales[('kernel',)])

In [16]: wn.scales
Out[16]:
{('kernel',): Param( # 1 (4 B)
   value=Array([1.], dtype=float32)
 )}

In [17]: nnx.state(wn)
Out[17]:
State({
  'layer_instance': {
    'bias': Param( # 1 (4 B)
      value=Array([0.], dtype=float32)
    ),
    'kernel': Param( # 3 (12 B)
      value=Array([[ 0.04756278],
             [-0.30357933],
             [ 0.782016  ]], dtype=float32)
    )
  },
  'scales': {
    ('kernel',): Param( # 1 (4 B)
      value=Array([1.], dtype=float32)
    )
  }
})

In [18]: nnx.state(wn, nnx.Param)
Out[18]:
State({
  'layer_instance': {
    'bias': Param( # 1 (4 B)
      value=Array([0.], dtype=float32)
    ),
    'kernel': Param( # 3 (12 B)
      value=Array([[ 0.04756278],
             [-0.30357933],
             [ 0.782016  ]], dtype=float32)
    )
  },
  'scales': {
    ('kernel',): Param( # 1 (4 B)
      value=Array([1.], dtype=float32)
    )
  }
})
```

### PR DESCRIPTION
**Pull request by jdymitarai**
### Summary

Fixes #5577.

In `nnx.WeightNorm`, `scales` was initialized with raw JAX arrays inside `nnx.data`, preventing them from being identified as `nnx.Param`. Consequently:
- `scales` was omitted from `nnx.state(model, nnx.Param)`
- `scales` did not receive gradients during backpropagation
- Optimizers could not update `scales`

This PR wraps the initialized scale arrays in `nnx.Param` inside `self.scales = nnx.data({...})`, allowing them to be tracked as learnable parameters as specified in the original weight normalization paper ([Salimans & Kingma, 2016](https://arxiv.org/abs/1602.07868)) and consistent with `flax.linen.WeightNorm`.

### Changes
- In `flax/nnx/nn/normalization.py`: wrap `init_scales(param)` with `nnx.Param(...)` in `WeightNorm.__init__`.
- In `tests/nnx/nn/normalization_test.py`: add `TestWeightNorm` verifying `scales` is an `nnx.Param`, appears in `nnx.state(model, nnx.Param)`, receives non-zero gradients via `nnx.grad`, and handles `use_scale=False`.

### Test Plan
- Ran all normalization tests:
  ```bash
  pytest tests/nnx/nn/normalization_test.py
  ```
  All 110 tests passed.

### CODE REVIEW

**Comment by jdymitarai** [comment]
Closing this PR to avoid review overhead on the team. The context remains in the thread for future reference if helpful. Thanks for your time.
